dozer-sql/src/pipeline/product/join/processor.rs

```diff
@@ -16,6 +17,28 @@ pub struct ProductProcessor {
 
 impl ProductProcessor {
     pub fn new(join_operator: JoinOperator) -> Self {
+        describe_gauge!(
+            "product.left_lookup_size",
+            "Total number of items in the left lookup table"
+        );
+        describe_gauge!(
+            "product.right_lookup_size",
+            "Total number of items in the right lookup table"
+        );
+        describe_gauge!(
+            "product.unsatisfied_joins",
```

Comment:
Looks like this can only increase? In that case we can use a counter

Reply:
lookup table can decrease in case of TTL eviction

Reply:
I'm talking about `unsatisfied_joins`